bench/bench_skiplist.ml

```diff
@@ -0,0 +1,57 @@
+open Saturn
+
+let workload num_elems num_threads add remove =
+  let sl = Skiplist.create () in
+  let elems = Array.init num_elems (fun _ -> Random.int 10000) in
+  let push () =
+    Domain.spawn (fun () ->
+        let start_time = Unix.gettimeofday () in
+        for i = 0 to (num_elems - 1) / num_threads do
+          Domain.cpu_relax ();
+          let prob = Random.float 1.0 in
+          if prob < add then Skiplist.add sl (Random.int 10000) |> ignore
+          else if prob >= add && prob < add +. remove then
+            Skiplist.remove sl (Random.int 10000) |> ignore
+          else Skiplist.mem sl elems.(i) |> ignore
+        done;
+        start_time)
+  in
+  let threads = List.init num_threads (fun _ -> push ()) in
+  let start_time_threads =
+    List.map (fun domain -> Domain.join domain) threads
+  in
+  let end_time = Unix.gettimeofday () in
+  let time_diff = end_time -. List.nth start_time_threads 0 in
```

Comment:
Hmm... The timing collection seems to just use the start time of the first domain and the time after joining all of the threads.

How about:

1. Use a barrier to synchronize all the domains before their loops.
2. Individually time the loop inside each domain and return that from each domain.

Various measures could then be calculated from the collection of timings.  E.g. compute average of the times from each domain to get roughly the same kind of measurement as here, but taking all domains into account rather than the start time of the first domain and the end time (+ some) of the domain that finished last.